themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js

```diff
@@ -965,6 +965,9 @@ module.controller('RealmIdentityProviderCtrl', function($scope, $filter, $upload
         for (var key in data) {
             $scope.identityProvider.config[key] = data[key];
         }
+        if (data["enabled"] != undefined) {
+            $scope.identityProvider.enabled = data["enabled"];
```

Comment:
```suggestion
        if (data["enabledFromMetadata"] !== undefined) {
            $scope.identityProvider.enabled = data["enabledFromMetadata"];

```
(Please note `!==` and not `!=`)

Reply:
Ok.
Moreover, I observed that if I import an enabled SAML IdP , enabled will be off although enabledFromMetadata will be true.
If I change onoffswitch attribute to onoffswitchvalue ( realm-identity-provider-saml.html line 39) , import will work correctly. However, when admin create or edit SAML Idp, enabled is always off. 
The problem is that identityProvider.enabled is initiated as boolean and data return string as value.
How should I write code correctly without changing app.js ( lines 2236-2366)?

Reply:
```suggestion
            $scope.identityProvider.enabled = data["enabledFromMetadata"] == "true";
```

Reply:
You are right. However, if must exists for other Identity Providers like OIDC. In these Identity Providers enabledFromMetadata will not exist in data.